extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSinkFactory.java

```diff
@@ -100,7 +100,7 @@ public DataSink createSink(DataFlowRequest request) {
 
         return S3DataSink.Builder.newInstance()
                 .bucketName(destination.getProperty(BUCKET_NAME))
-                .keyName(destination.getKeyName())
+                .keyName(request.getSourceDataAddress().getKeyName())
```

Comment:
can this new behavior be tested?

Reply:
@ndr-brt yes, I have tested it. I transferred file from one AWS S3 bucket to another AWS S3 bucket and it was working fine. File name is copied with the actual name and extension. I think that the method that we were using before will not work. In the previous changes that I had done, we were putting the filename in the keyName field and we sent it from consumer side but in the code the secrets are also stored by the keyName so if there are 2 consumers and they send the request for the same asset at the same time their secrets will overwrite each other. I rolled back those changes and now I simply changes the keyName to the keyName present in the sourceDataAddress Object before copying it to AWS S3 bucket.

Reply:
@MUsman1992 I mean, with an automated test, otherwise we can't ensure that this will work effectivly.